File: sdks/javascript/examples/sprite-demo.ts
/**
 * 2D Sprite Demo
 * 
 * Demonstrates 2D sprite rendering with the Windjammer TypeScript SDK.
 * 
 * Run with: npm run build && node dist/examples/sprite-demo.js
 */

import { App, Vec2, Sprite, Camera2D } from '../src/index';

console.log('=== Windjammer 2D Sprite Demo (TypeScript) ===');

// Create 2D application
const app = new App();

// Setup system
app.addStartupSystem(() => {
  console.log('\n[Setup] Creating 2D scene...');
  
  // Create camera
  const camera = new Camera2D(new Vec2(0, 0), 1.0);
  console.log(`  - ${camera}`);
  
  // Create sprites
  const sprite1 = new Sprite({
    texture: 'player.png',
    position: new Vec2(0, 0),
    size: new Vec2(64, 64)
  });
  console.log(`  - ${sprite1}`);
  
  const sprite2 = new Sprite({
    texture: 'enemy.png',
    position: new Vec2(100, 100),
    size: new Vec2(48, 48)
  });
  console.log(`  - ${sprite2}`);
  
  console.log('[Setup] Scene ready!');
});

// Update system
app.addSystem(() => {
  // This would rotate sprites each frame
});

console.log('2D application configured!');
console.log('- Camera: Orthographic');
console.log('- Sprites: Enabled');
console.log('- Physics: 2D');
console.log();

// Run the application
app.run();

File: sdks/cpp/examples/sprite_demo.cpp
/**
 * 2D Sprite Demo
 * 
 * Demonstrates 2D sprite rendering with the Windjammer C++ SDK.
 * 
 * Build with: cmake --build build && ./build/examples/sprite_demo
 */

#include <windjammer/windjammer.hpp>
#include <iostream>

using namespace windjammer;

int main() {
    std::cout << "=== Windjammer 2D Sprite Demo (C++) ===" << std::endl;
    
    // Create 2D application
    App app;
    
    // Setup system
    app.add_startup_system([]() {
        std::cout << "\n[Setup] Creating 2D scene..." << std::endl;
        
        // Create camera
        auto camera = Camera2D{Vec2{0.0f, 0.0f}, 1.0f};
        std::cout << "  - " << camera << std::endl;
        
        // Create sprites
        auto sprite1 = Sprite{
            .texture = "player.png",
            .position = Vec2{0.0f, 0.0f},
            .size = Vec2{64.0f, 64.0f}
        };
        std::cout << "  - Sprite 'player.png' at (0, 0) size=(64, 64)" << std::endl;
        
        auto sprite2 = Sprite{
            .texture = "enemy.png",
            .position = Vec2{100.0f, 100.0f},
            .size = Vec2{48.0f, 48.0f}
        };
        std::cout << "  - Sprite 'enemy.png' at (100, 100) size=(48, 48)" << std::endl;
        
        std::cout << "[Setup] Scene ready!" << std::endl;
    });
    
    // Update system
    app.add_system([]() {
        // This would rotate sprites each frame
    });
    
    std::cout << "2D application configured!" << std::endl;
    std::cout << "- Camera: Orthographic" << std::endl;
    std::cout << "- Sprites: Enabled" << std::endl;
    std::cout << "- Physics: 2D" << std::endl;
    std::cout << std::endl;
    
    // Run the application
    app.run();
    
    return 0;
}

